invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/modelSelected.ts

```diff
@@ -78,3 +94,138 @@ export const addModelSelectedListener = (startAppListening: AppStartListening) =
     },
   });
 };
+
+const handleReferenceImageModelSwitching = (
+  state: RootState,
+  dispatch: AppDispatch,
+  newModel: MainModelConfig,
+  log: typeof logger
+) => {
+  // Get all available models from the models query
+  const modelsQueryResult = modelsApi.endpoints.getModelConfigs.select()(state);
+  if (!modelsQueryResult.data) {
+    return;
+  }
+
+  const allModels = modelConfigsAdapterSelectors.selectAll(modelsQueryResult.data);
```

Comment:
Use the existing selector in `invokeai/frontend/web/src/services/api/hooks/modelsByType.ts` to get all IP Adapter models.